mach_eval/analyzers/electromagnetic/winding_factors.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+
+class WindingFactorsProblem:
+    """Problem class for winding factor analyzer
+    Attributes:
+        harmonics: number of harmonics to be included in calculations []
+        winding_layout: array of winding layers beginning with top layer
+                        and ending with bottom layer, length will be number 
+                        of teeth and height will be number of layers, "+" 
+                        for conductor into page, 0 for no present conductors, 
+                        and "-" for conductor out of page [turns]
+        alpha_1: angle of first slot from +x axis [rad]
+    """
+    
+    def __init__(self, harmonics, winding_layout, alpha_1):
+        
+        self.harmonics = harmonics
+        self.winding_layout = winding_layout
+        self.alpha_1 = alpha_1
+        
+        
+            
+class WindingFactorsAnalyzer:
+    """Analyzer class to evaluate winding factors"""
+    
+    
+    
+    def analyze(self, problem="WindingFactorsProblem"):
+        """Determines winding factors using calculations function
+        
+        Args:
+            harmonics: harmonics array definied in problem
+            winding_layout: winding layout defined in problem
+            alpha_1: angle of slot 1 defined in problem
+            
```

Comment:
Instead of specifying the arguments of the problem class's initializer, specify the arguments of the `analyze()` function here (this should be just the problem class)